x-pack/plugins/security_solution/public/detections/components/rules/eql_query_bar/eql_query_bar.tsx

```diff
@@ -52,9 +70,15 @@ export const EqlQueryBar: FC<EqlQueryBarProps> = ({
 }) => {
   const { addError } = useAppToasts();
   const [errorMessages, setErrorMessages] = useState<string[]>([]);
-  const { isValidating, setValue } = field;
+  const {
+    isValidating,
+    value: fieldValue,
+    setValue: setFieldValue,
+  } = field as FieldHook<FieldValueQueryBar>;
```

Comment:
is casting needed here? `field` already has type `  field: FieldHook<DefineStepRule['queryBar']>;` which seems the same as casted

Reply:
good point!